### PR TITLE
Delete coverage data source

### PR DESCRIPTION
**Pull request by nberard**

### CODE REVIEW

**Comment by buildbotctp** [comment]
Can one of the admins verify this patch?

**Comment by nberard** [comment]
jenkins, test this please

File: tartare/core/models.py [hunk 1]
```diff
@@ -82,18 +86,17 @@ def save_grid_calendars(self, file):
...
+        # when we delete the file all process reading it will get invalid data
+        # TODO: We will need to implement a better solution
         delete_file_from_gridfs(self.grid_calendars_id, gridfs=gridfs)
         self.grid_calendars_id = id
 
     def get_grid_calendars(self):
         if not self.grid_calendars_id:
             return None
-        gridfs = GridFS(mongo.db)
+        GridFS(mongo.db)
```
Comment (antoine-de) [inline tartare/core/models.py]: you sure you need this ?

Comment (nberard) [inline tartare/core/models.py]: just unused var :p

Comment (antoine-de) [inline tartare/core/models.py]: but then the line does not seems very usefull :wink: (especially since it is done in the function below)

Comment (nberard) [inline tartare/core/models.py]: so either remove GridFS(mongo.db) or call return get_file_from_gridfs(self.grid_calendars_id, gridfs)

File: tartare/interfaces/coverage_data_source_subscription.py [hunk 2]
```diff
@@ -65,3 +66,14 @@ def post(self, coverage_id):
             return {'message': str(e)}, 400
 
         return {'coverages': schema.CoverageSchema().dump([coverage], many=True).data}, 200
+
+    def delete(self, coverage_id, data_source_id):
+        coverage = models.Coverage.get(coverage_id)
+        if coverage is None:
+            return {'message': 'Bad coverage {}.'.format(coverage_id)}, 404
```
Comment (antoine-de) [inline tartare/interfaces/coverage_data_source_subscription.py]: I don't know how you want to procede, but you should also raise exceptions like https://github.com/CanalTP/tartare/pull/81 (but we can merge both PR before you make the changes)

File: tartare/core/models.py [hunk 3]
```diff
@@ -147,6 +149,11 @@ def has_data_source(self, data_source):
     def add_data_source(self, data_source):
         self.data_sources.append(data_source.id)
 
+    def remove_data_source(self, coverage_id, data_source_id):
+        if data_source_id in self.data_sources:
```
Comment (datanel) [inline tartare/core/models.py]: you already do it in the controller

File: tartare/core/models.py [hunk 4]
```diff
@@ -147,6 +149,11 @@ def has_data_source(self, data_source):
...
+            self.data_sources.remove(data_source_id)
+            self.update(coverage_id, {"data_sources": self.data_sources})
```
Comment (datanel) [inline tartare/core/models.py]: you don' t need coverage_id as you always have in self object
`self.update(self.id, {"data_sources": self.data_sources})`

File: tartare/interfaces/coverage_data_source_subscription.py [hunk 5]
```diff
@@ -65,3 +66,14 @@ def post(self, coverage_id):
...
+
+        if data_source_id not in coverage.data_sources:
+            return {'message': 'Unknown data_source_id attribute in uri.'}, 404
+
+        coverage.remove_data_source(coverage_id, data_source_id)
```
Comment (datanel) [inline tartare/interfaces/coverage_data_source_subscription.py]: you shoud try exept PyMongoError, ValueError if for some reasons there is a connection to mongo that failed for example

Comment (nberard) [inline tartare/interfaces/coverage_data_source_subscription.py]: i can let it raise so we catch it top level and return a 500 response no?

File: documentation/openAPI.yaml [hunk 6]
```diff
@@ -249,12 +249,32 @@ paths:
...
+          required: true
+        - in: path
+          name: data_source_id
+          type: string
+          description: Identifier of the data_source that needs to be removed from the coverage
+          required: true
+      responses:
+        204:
+          description: When data source {data_source_id} is correctly deleted from coverage
+        404:
```
Comment (datanel) [inline documentation/openAPI.yaml]: you can use
```
404:
          description: Resource not found
          schema :
            $ref: '#/definitions/error_404'
```

**Pull request merged**